src/Integration.Vsix.UnitTests/InfoBar/InfoBarManagerTests.cs

```diff
@@ -33,6 +33,12 @@
 
 namespace SonarLint.VisualStudio.Integration.UnitTests
 {
+    // NB we need to use the ConfigurableServiceProvider for these tests because it correctly handles
+    // COM type equivalance for embedded interop types such as SVsInfoBarUIFactory.
+    // Otherwise, we can get unexpected failures such as tests passing in the VS2022 build but not in
+    // the VS2019 build
+    // See https://learn.microsoft.com/en-us/dotnet/framework/interop/type-equivalence-and-embedded-interop-types
+
```

Comment:
This caused a lot of problems. I eventually fixed it by changing the new tests to use the old `ConfigurableXXX` classes, rather than using `Mock<x>`.

Reply:
Yea, it is a bit unfortunate that we have to deal with it like this but atleast it works now!